### DISCUSSION TITLE
equivalent of Cypress.env inside cypress.config.ts?

### DISCUSSION THREAD

**Discussion by dhorkin** [cypress-io/cypress#27440] [Questions and Help]
inside my `cypress.config.ts`, I'm effectively wanting to do this:

```ts
export dfeault defineConfig({
    e2e: {
        env: {
            something: `foo.${Cypress.env('bar')}`
        }
    }
});
```

But `Cypress.env` doesn't appear usable there. I could do this:
```ts
import { bar as barFromConfig } from './cypress.env.json';
``` 
but then if the environment variable `CYPRESS_bar` is defined, I would miss it.

I could combine the two:
```ts
something: `foo.${process.env.CYPRESS_bar ?? barFromConfig}`
```
but this feels a bit hacky and would miss if it was overridden via `--env`, etc.

I did notice that `e2e.setupNodeEvents(on, config)` has a `config.env`, but it appears changing the value there doesn't persist. Trying to store a local variable of the value of `config.env` doesn't work either, since setupNodeEvents is called after defineConfig

Short of reinventing the wheel / precedence order, is there a better option?

**Comment by kasir-barati** [comment]
Did you find any good enough satisfying answer for this question? Because I am also thinking to use `process.env` but not sure completely.

**Reply by dhorkin** [reply]
I ended up using external libraries to fake it. Wasn't really thrilled with this solution, but it works.

```ts
import { defineConfig } from 'cypress';

import { environment } from './cypress.env.json';

import dotenv from 'dotenv';
import { expand as dotenvExpand } from 'dotenv-expand';

export default defineConfig({
    e2e: {
        env: {
            env: {
                hideCredentials: true,
                // Read in environment variables from dotenv (i.e. the .env file)
                // The dotenvExpand allows the .env to reference other environment variables with $
                ...dotenvExpand(dotenv.config({ path: `.env.${process.env.CYPRESS_environment ?? environment}` })).parsed,
        },
    }
})
```
